quantum/split_common/matrix.c

```diff
@@ -55,13 +52,36 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
     static bool debouncing = false;
 #endif
 
+#if defined(USE_I2C) || defined(EH)
+
 #if (MATRIX_COLS <= 8)
 #    define print_matrix_header()  print("\nr/c 01234567\n")
 #    define print_matrix_row(row)  print_bin_reverse8(matrix_get_row(row))
 #    define matrix_bitpop(i)       bitpop(matrix[i])
 #    define ROW_SHIFTER ((uint8_t)1)
 #else
 #    error "Currently only supports 8 COLS"
```

Comment:
I'm not sure if it helps here, or it's something that you'd want to look into later ...

But the Orthodox keyboard supports 9 columns over serial and I2C.  It may be worth checking out it's code and pulling the fix there and porting it over.

Reply:
For now, I think `#if` is necessary.

 Since this pull-request deals with the change of serial.c, I think that i2c.c should maintain the current state without changing it. .

Reply:
Sounds good!